File: CarePackage/Software/Parsec.cs
namespace CarePackage.Software;

public class Parsec : BaseSoftware
{
    public override string           Key            { get; set; } = "parsec";
    public override string           Name           { get; set; } = "Parsec";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Gaming;
    public override string           DownloadName   { get; set; } = "parsec-windows.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = true;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.parsec;
    public override string           Homepage       { get; set; } = "https://parsec.app";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://builds.parsec.app/package/parsec-windows.exe");
}

File: packages/app/src/services/software/definitions/parsec.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'parsec';
	public name = 'Parsec';
	public category = [SoftwareCategory.Gaming];
	public downloadName = 'parsec-windows.exe';
	public requiresAdmin = true;
	public icon = 'parsec.png';
	public homepage = 'https://parsec.app';

	public async resolveDownloadUrl() {
		return ok('https://builds.parsec.app/package/parsec-windows.exe');
	}
}
